server.go

```diff
@@ -35,31 +35,32 @@ func (c Context) Check(msg string, err error) error {
 	return nil
 }
 
-// EmptyResonse writes a 204 No Content response. This should be used when the
-// client should not expect a response body.
-func (c Context) EmptyResonse() {
-	c.ResponseWriter.WriteHeader(http.StatusNoContent)
-}
-
-// Encode writes the JSON encoding of v to the response body.
-func (c Context) Encode(v interface{}) {
-	c.ResponseWriter.Header().Set("Content-Type", "application/json")
-	// encode nil slices as [] and nil maps as {} (instead of null)
-	if val := reflect.ValueOf(v); val.Kind() == reflect.Slice && val.Len() == 0 {
-		c.ResponseWriter.Write([]byte("[]\n"))
-		return
-	} else if val.Kind() == reflect.Map && val.Len() == 0 {
-		c.ResponseWriter.Write([]byte("{}\n"))
-		return
+// Encode writes the encoding of v to the response body. If v implements the
+// ResponseWriter interface, it is written directly.
+// Otherwise, it is marshalled as JSON.
```

Comment:
whoops, missed this -- need to update the comment to reflect the removal of `ResponseWriter` (also the `nil` special case)